Sudoku/errorChecking.js

```diff
@@ -0,0 +1,81 @@
+const checkSquare = (squareRepeat, squaresArr) => {
+  for(let bigSquare of squaresArr) {
+    for(let iterI in bigSquare){
+      iterI = parseInt(iterI)
+      for(let iterJ = iterI+1; iterJ < 9; iterJ++) { // What big O is this?
+        if( bigSquare[ iterI ] === bigSquare[ iterJ ] && bigSquare[ iterI ] !== '.' ) {
+          squareRepeat = true
+          break
+        }
+      }
+    }
+  }
+
+  return squareRepeat
+}
+
+const checkContainer = container => {
+  for(let itI of container) {
+    const frequencyCount = {}
+    for(let itJ of itI) {
+      if(frequencyCount[itJ] && itJ !== '.'){
+        return true
+      } else {
+        frequencyCount[itJ] = 1
+      }
+    }
+  }
+
+  return false
+}
+
+const checkAnyNumbersAvailable = (rowsArr, columnsArr) => {
+  for(let itPR in rowsArr) {
+    for(let itPC in columnsArr) {
+      const pointRow = itPR
+      const pointCol = itPC
+      const usedNumbers = {}
+      let freeNumbersAvailable = false
+      for(let itR of rowsArr[ pointRow ]) {
+        usedNumbers[ itR ] = true
+      }
+      for(let itC of columnsArr[ pointCol ]) {
+        usedNumbers[ itC ] = true
+      }
+      for(let itI = 1; itI <= 9; itI++) {
```

Comment:
Something that has been suggested to me multiple times is to define constant values as variables. On line 45 the number 9 obviously holds some significance, but as a reviewer its not exactly clear why. Why 9? If it was stored as a const with a name like const MAX_THINGS_ALLOWED = 9 it would help make things a little more clear.